tests/test_0603-dask-delayed-open.py

```diff
@@ -38,3 +38,29 @@ def test_multiple_delay_open():
         == true_val["px1"]
     )
     assert arr.all()
+
+
+@pytest.mark.parametrize("open_files", [False, True])
+def test_supplied_chunks(open_files):
+    filename1 = skhep_testdata.data_path("uproot-Zmumu.root")
+    filename2 = skhep_testdata.data_path("uproot-Zmumu-uncompressed.root")
+    true_val = uproot.concatenate([filename1 + ":events", filename2 + ":events"])
+
+    chunks1 = [0, 2305]
+    chunks2 = [[0, 2305]]
+
+    files = {
+        filename1: {"object_path": "events", "chunks": chunks1},
+        filename2: {"object_path": "events", "chunks": chunks2},
+    }
+
+    assert uproot.dask(files, open_files=open_files)["px1"].divisions == (
+        None,
+        None,
+        None,
+    )
```

Comment:
I'm confused about this. Although `open_files=False` would ordinarily give you unknown divisions, specifying them explicitly should make them known. Isn't that the whole point of this PR?

(What happens when the files are eventually do get opened and the chunks are impossible because they go beyond the number of entries in the file is another question—maybe it should raise an error, maybe it should trim like a Python `slice`, etc.)

This is weird: they're never set!

```python
>>> filename1 = skhep_testdata.data_path("uproot-Zmumu.root")
>>> filename2 = skhep_testdata.data_path("uproot-Zmumu-uncompressed.root")
>>> uproot.dask([filename1, filename2], open_files=True)["px1"].divisions
(None, None, None)
>>> uproot.dask([filename1, filename2], open_files=False)["px1"].divisions
(None, None, None)
```

And yet the property has meaning:

```python
>>> import awkward as ak
>>> import numpy as np
>>> import dask_awkward as dak
>>> a = dak.from_awkward(ak.Array(np.arange(100)), npartitions=3)
>>> a
dask.awkward<from-awkward, npartitions=3>
>>> a.divisions
(0, 34, 68, 100)
```

The `open_files=False` case is sensitive to the number of chunks:

```python
>>> uproot.dask({filename1: {"object_path": "events", "chunks": [0, 2305]}}, open_files=False)["px1"].divisions
(None, None)
>>> uproot.dask({filename1: {"object_path": "events", "chunks": [0, 1000, 2305]}}, open_files=False)["px1"].divisions
(None, None, None)
>>> uproot.dask({filename1: {"object_path": "events", "chunks": [0, 1000, 2000, 2305]}}, open_files=False)["px1"].divisions
(None, None, None, None)
```

and

```python
>>> uproot.dask({filename1: {"object_path": "events", "chunks": [[0, 2305]]}}, open_files=False)["px1"].divisions
(None, None)
>>> uproot.dask({filename1: {"object_path": "events", "chunks": [[0, 1000], [1000, 2305]]}}, open_files=False)["px1"].divisions
(None, None, None)
>>> uproot.dask({filename1: {"object_path": "events", "chunks": [[0, 1000], [1000, 2000], [2000, 2305]]}}, open_files=False)["px1"].divisions
(None, None, None, None)
```

The `open_files=True` case isn't, but that's my job.

I noticed that this is already a case in which the `chunks` are asking for more events than are available. The first file has only 2304 events, so the whole range would be `[0, 2304]`. It doesn't complain about too-large values:

```python
>>> uproot.dask({filename1: {"object_path": "events", "chunks": [[0, 5000]]}}, open_files=False)["px1"].compute()
<Array [-41.2, 35.1, 35.1, 34.1, ..., 32.4, 32.4, 32.5] type='2304 * float64'>
```

I _suspect_ that you'll want it to raise errors. If you're providing exact numbers of entries, I think you'd want to know if the numbers are wrong.

Reply:
It shouldn't be too hard to add in the divisions in this case, even if we haven't opened the files. I think that's a constructor argument for the dask array, or we can set them?

I think there's a huge and useful distinction between hinting the chunks (when you're sure your files won't change underneath you) and needed to calculate them from the files.

In particular this is useful because it is not *always* the case that we can uniformly split files into chunks, and if we want ~uniform processing time per-partition we need to be able to say in what way we want chunks.

Reply:
Suppose that a chunk's `start` and `stop` are both greater than the number of entries. That partition would be empty. Guessing wrong and getting lots of empty partitions isn't a problem, is it?

Reply:
In terms of the system complaining about things not lining up. 

The behavior in steps_per_file mode does tail clipping, but I guess we're expecting the user to supply something truly exact there.  Sure, that makes sense to me.

Reply:
I'll leave it as clipping, rather than raising errors.

Reply:
Change of plan (after talking on Slack): if any user-provided steps (i.e. "chunks") go out of bounds for any TTree, it raises an error.

This strictness has a benefit—if we know that we'll get the entry ranges we ask for or die trying, the Dask array `chunks` or dask-awkward `divisions` can be known before `.compute()`. As a consequence, there will be fewer `unknown_lengths` in type-tracer arrays, so unknown values propagate less, and downstream situations that need known lengths or values won't have to give up. It may be that some workflow is impossible with `open_files=False` and no steps in `files`, but possible with them (which is a nice incentive).